Testing/update sensor configurations

Blade-iii opened the pull request:
### Testing for user story #13 

#### What's been tested
- ```LoadConfig``` method
-  ```ConvertStringToFloat``` method

#### LoadConfig Tests
- Tests the base functionality of the method - which checks if the data is being populated into the collections from the database provided.
- Tests the scenario there is no data in the database - by providing an empty list to the database

#### CovertStringToFloat Tests
- Tests the base functionality of the method - providing a string and passing it into the method and check if the type it outputs is a float.
- Test invalid format provided to the method - by giving a string which is text instead of a numeric value and checks if the try catch is triggered and returns 0.
- Tests null provided to the method - by giving null to a string and passing it into the method to check if the method can handle null values.

> [!NOTE]
> Updated ```FakeDatabaseService``` to simulate an empty database with no values.
> ```UpdateConfigCommand``` this is not tested at the moment as it's a private method but it would be good to test.

Blade-iii committed:
Create UpdateSensorTests.cs

Create the test and import libraries needed for testing

Blade-iii committed:
ConvertStringToFloat Tests

Creating tests for this method to test if it can handle incorrect input formats

Blade-iii committed:
Added Null check and test

Blade-iii committed:
LoadConfig Test

Blade-iii committed:
Refactoring - Dependency injection

The view model is no longer dependant on the sql database and it can now mock with the fakedb service

Blade-iii committed:
Refactoring

Adding SetItems method to allow testing of if the database is down or has no data. So i can test empty.

Refactored the tests to actually use the method

NikitaEdin commented on REA/ViewModels/UpdateSensorViewModel.cs:
```diff
@@ -179,7 +191,7 @@ public async Task UpdateConfig()
                 item.MaxMeasurement = ConvertStringToFloat(sensorMaxValue);
```
Missing null checks for `sensorMinValue` and `sensorMaxValue`, resulting in null exceptions, which should be checked before calling `ConvertStringToFloat`.
The null exceptions are caught up in the StringToFloat, but they should be tested before reaching that point.

**Suggestion:**
```csharp
 if (string.IsNullOrWhiteSpace(sensorMinValue) || string.IsNullOrWhiteSpace(sensorMaxValue)) {
     return;
 }
```

NikitaEdin commented on REA.Tests/ViewModels/UpdateSensorTests.cs:
```diff
@@ -0,0 +1,157 @@
+            var viewModel = new UpdateSensorViewModel(fakeDb);
+
+            float convertedFloat;
+            string input = null;
```
Setting a null to a non-nullable type is a bad practice and should be avoided.
Instead, use `string.empty`.

And the same is true for `Assert.NotNull(convertedFloat);` as float can't be nullable unless strictly defined as a nullable primitive type.
Consider removing this assert:
https://github.com/NikitaEdin/SET09102-2025-TR2/blob/e4382178a40da1e735f6c5539d4147a50e5f9917/REA.Tests/ViewModels/UpdateSensorTests.cs#L152

> [!TIP]
>  All primitive data types have default values even if nothing is explicitly set into them.
> This means that `float convertedFloat;` will have `0` in it by default.

NikitaEdin reviewed (changes requested):
Great work on the unit tests and lowering the class coupling with the database in the ViewModel.

However, there are 2 changes that are highly suggested to be revised before merging to dev.
- Null checks in updating a selected sensor in `UpdateSensorViewModel.cs`.
- Removing bad-practice use of nulls in `ConvertStringToFloatTest_Null` in `UpdateSensorTests.cs`.

Blade-iii committed:
Added Suggested changes

Removed Null check and used string.empty and also added null check for the user inputs

NikitaEdin reviewed (approved):
All previous feedback has been addressed, and the changes are well-implemented and consistent with the requested updates. 

Well done!
